internal/provisioner/kops_provisioner_cluster.go

```diff
@@ -556,7 +556,7 @@ func (provisioner *KopsProvisioner) ProvisionCluster(cluster *model.Cluster, aws
 
 	prom, _ := k8sClient.GetNamespace("prometheus")
 
-	if prom != nil {
+	if prom.Name != "" {
```

Comment:
We do not do error check from `k8sClient.GetNamespace` we also do not do `nil` check, so if `prom` ends up being `nil` this will panic.

Reply:
The problem with the previous code is that prom is never nil as prom returned empty object instead of `nil`, this caused in rare cases where prometheus utility was not installed in a cluster sloth always to try to get installed while it can't as it sits in prometheus namespace. Current code fix this code as `prom.Name` can be either `prometheus` or empty string. @Szymongib

Reply:
Yeah that makes sense but we still should do nil check since `GetNamespace` returns a pointer. The implementation of `GetNamespace` might change in the future where the `nil` will be returned instead of empty object which will lead to panic here. So something like `if prom != nil && prom.Name != "" {` should work.

Reply:
Yes that makes sense